watertap/tools/analysis_tools/loop_tool/tests/test_data_merging_tool.py

```diff
@@ -0,0 +1,119 @@
+#################################################################################
+# WaterTAP Copyright (c) 2020-2023, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National Laboratory,
+# National Renewable Energy Laboratory, and National Energy Technology
+# Laboratory (subject to receipt of any required approvals from the U.S. Dept.
+# of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+import pytest
+import os
+import numpy as np
+
+from watertap.tools.analysis_tools.loop_tool import (
+    data_merging_tool,
+)
+import h5py
+
+__author__ = "Alexander V. Dudchenko (SLAC)"
+
+_this_file_path = os.path.dirname(os.path.abspath(__file__))
+
+
+def create_test_h5_file():
+    f_name = _this_file_path + "/test_h5_file.h5"
```

Comment:
Using tmp_path causes issues as it tries to create same directory multiple times (I get 'directory exists errors). I also want to be able to access the directory in subsequent test, so using a global path works better. This is the same approach taken for testing LSRRO multi-sweep data generation. If this starts causing issues, it can be changed in a different pr.